### PR TITLE
fix: tolerate empty-string entries in acceptance-report string arrays

### PR DESCRIPTION
**Pull request by hjiang**
## Problem

Chain subagents (e.g. the `scout` agent in a scout → planner → worker → reviewer chain) occasionally end their structured `acceptance-report` with `[""]` for a field that has no items. The report parser (`validateStringArrayField`) requires every entry in a string-array field to be a non-empty string, so **one placeholder empty-string entry rejects the entire report** — a run that exited 0 with complete, correct findings is marked `rejected` (runtime check `attestation` fails with e.g. `validationOutput[0]: expected non-empty string; got ""`).

## Fix

Two small changes in `src/runs/shared/acceptance.ts`:

1. **Parse-time tolerance** — `normalizeAcceptanceReportValue` now drops empty/whitespace-only entries from the five string-array fields (`changedFiles`, `testsAddedOrUpdated`, `validationOutput`, `residualRisks`, `reviewFindings`). Non-string entries are **kept**, so structural garbage (objects, numbers) still fails validation exactly as before.
2. **Contract clarity** — `formatAcceptancePrompt` now tells the model: *"An empty-string entry (\`[""]\`) is invalid and fails the report — use \`[]\` when nothing applies."*

## Tests

- Empty-string entries now parse to `[]` for all five fields.
- Mixed arrays (`["real finding", ""]`) keep only meaningful entries.
- Single-string coercion (`changedFiles: "src/file.ts"`) is unaffected.
- Non-string entries (`reviewFindings: [{}]`) are still rejected with `expected non-empty string; got object`.
- Updated the existing "blank evidence" rejection case to use a non-string entry, since `[""]` is now tolerated by design.

`test/unit/acceptance.test.ts`: 46/46 pass (`node --experimental-strip-types --test test/unit/acceptance.test.ts`).

## Motivation

Real-world failure: a read-only scout ran cleanly (exit 0, full findings with file paths and line ranges) but was rejected purely because its optional `validationOutput` field contained `[""]`. The tolerance preserves the validator's job (reject garbage) while not punishing models for using an empty string to mean "no items".

### CODE REVIEW

File: src/runs/shared/acceptance.ts [hunk 1]
```diff
@@ -433,6 +433,7 @@ export function formatAcceptancePrompt(acceptance: ResolvedAcceptanceConfig, opt
 		"",
 		"Finish with a fenced JSON block tagged `acceptance-report` in this shape:",
 		"Use empty arrays when no items apply; array fields contain strings unless object entries are shown.",
+		"An empty-string entry (`[\"\"]`) is invalid and fails the report — use `[]` when nothing applies.",
```
[the remark this answers is not in the document]
Comment (hjiang) [reply src/runs/shared/acceptance.ts]: Fixed in d51824f: reworded the contract prompt to match the parser's actual behavior — now reads "Empty-string entries (`[""]`) are ignored; use `[]` when nothing applies." (adopted your suggested wording). Also updated the corresponding test assertion to be case/plural-aware (`/empty-string entr(?:y|ies)/i`) and synced the local re-apply patch in my config repo so the installed package and upstream stay identical.

File: src/runs/shared/acceptance.ts [hunk 2]
```diff
@@ -612,9 +613,17 @@ function normalizeAcceptanceReportValue(value: unknown, pathLabel = ""): { value
...
-				normalized[canonical] = typeof fieldValue === "string" ? [fieldValue] : fieldValue;
+			case "reviewFindings": {
+				// Tolerate empty-string entries ("[\"\"]"): models write them for "no items"
+				// and a single empty entry must not reject the whole report. Drop them at
+				// parse time; non-string entries are kept so validateStringArrayField still
+				// flags structural garbage.
+				const items = typeof fieldValue === "string" ? [fieldValue] : fieldValue;
+				normalized[canonical] = Array.isArray(items)
+					? items.filter((item) => typeof item !== "string" || item.trim().length > 0)
+					: items;
```
[the remark this answers is not in the document]
Comment (hjiang) [reply src/runs/shared/acceptance.ts:624]: Changelog gate: addressed in d51824f — added the entry under `## [Unreleased]` → `### Fixed`: "Tolerate empty-string entries in acceptance-report string-array fields instead of rejecting the whole report. Thanks to @hjiang for #1015."

CI gate: this is a fork PR, so the `Tests` workflow (`.github/workflows/test.yml`) requires a maintainer to approve the workflow run before it executes — GitHub has only run Greptile Review on head `d51824f`. I mirrored the CI steps locally on the head commit instead: `npm ci --ignore-scripts` ✓, `npm run typecheck` (tsc --noEmit) ✓, `npm test` → 1778 passed / 0 failed (8 cancelled, identical to pristine main). Please approve the workflow run (or merge) — nothing further is actionable from the fork side.
